### PR TITLE
[JXR-170] NullPointerException while parsing Java 15 multi-line String source

### PR DESCRIPTION
**Pull request by sman-81**
This PR offers a fix for bug [JXR-170]. Summary of changes:

- Add support for parsing sources with Java 15 multi-line Strings
- Add JUnit test case
- Implement toString on class BaseType
- Require non-null name on BaseType construtor
- Prevent construction of types with null name
- Add version to maven-enforcer-plugin plugin to prevent failure of integration test JXR-143_nofork

Following this checklist to help us incorporate your contribution quickly and easily:

 - [x] Make sure there is a [JIRA issue](https://issues.apache.org/jira/browse/JXR) filed 
       for the change (usually before you start working on it).  Trivial changes like typos do not 
       require a JIRA issue.  Your pull request should address just this issue, without 
       pulling in other changes.
 - [x] Each commit in the pull request should have a meaningful subject line and body.
 - [x] Format the pull request title like `[JXR-XXX] - Fixes bug in ApproximateQuantiles`,
       where you replace `JXR-XXX` with the appropriate JIRA issue. Best practice
       is to use the JIRA issue title in the pull request title and in the first line of the 
       commit message.
 - [x] Write a pull request description that is detailed enough to understand what the pull request does, how, and why.
 - [x] Run `mvn clean verify` to make sure basic checks pass. A more thorough check will 
       be performed on your pull request automatically.
 - [x] You have run the integration tests successfully (`mvn -Prun-its clean verify`).

If your pull request is about ~20 lines of code you don't need to sign an
[Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf) if you are unsure
please ask on the developers list.

To make clear that you license your contribution under 
the [Apache License Version 2.0, January 2004](http://www.apache.org/licenses/LICENSE-2.0)
you have to acknowledge this by using the following check-box.

 - [x] I hereby declare this contribution to be licenced under the [Apache License Version 2.0, January 2004](http://www.apache.org/licenses/LICENSE-2.0)

 - [ ] In any other case, please file an [Apache Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

### CODE REVIEW

**Review by slawekjaranowski — CHANGES_REQUESTED** [review]

File: maven-jxr/src/test/java/org/apache/maven/jxr/pacman/JavaFileImplTest.java [hunk 1]
```diff
@@ -46,4 +46,14 @@ public void testJXR_135_lotsOfNested() throws IOException
...
\ No newline at end of file
+    @Test
+    public void testJXR_170_multiLineString() throws IOException
+    {
+        JavaFileImpl javaFile = new JavaFileImpl( Paths.get(
+                "src/test/resources/jxr170/org/apache/maven/jxr/pacman/ClassWithMultiLineString.java" ),
+                "UTF-8" );
+        assertEquals( 1, javaFile.getClassTypes().size() );
+        assertEquals( "ClassWithMultiLineString", javaFile.getClassTypes().get(0).getName() );
+        assertEquals( "[ImportType[name=java.lang.*]]", javaFile.getImportTypes().toString() );
```
[the remark this answers is not in the document]
Comment (sman-81) [reply maven-jxr/src/test/java/org/apache/maven/jxr/pacman/JavaFileImplTest.java:57]: What do you mean?
getImportTypes returns a Set, getClassTypes returns a List (this is the original code unchanged by me)

File: maven-jxr-plugin/src/it/JXR-143_nofork/pom.xml [hunk 2]
```diff
@@ -44,6 +44,7 @@ under the License.
       <plugin>
         <groupId>org.apache.maven.plugins</groupId>
         <artifactId>maven-enforcer-plugin</artifactId>
+        <version>3.1.0</version>
```
[the remark this answers is not in the document]
Comment (sman-81) [reply maven-jxr-plugin/src/it/JXR-143_nofork/pom.xml:47]: Are you aware this integration test `JXR-143_nofork` fails on master as well?
The project requires that the integration tests run successfully (`mvn -Prun-its clean verify`).
So I fixed it.

The root cause is failure to find `maven-enforcer-plugin`.
After adding a plugin version, the test is green.

I've looked at the test however: the usage of maven-enforcer-plugin looks like non-sense, rules = AlwaysPass ;)

File: maven-jxr/src/main/java/org/apache/maven/jxr/pacman/BaseType.java [hunk 3]
```diff
@@ -36,22 +48,15 @@
...
-     * Set the name for this type
-     *
-     * @param name The new name value
-     */
-    public void setName( String name )
+    @Override
+    public String toString()
     {
-        this.name = name;
+        return getClass().getSimpleName() + "[name=" + name + "]";
```
[the remark this answers is not in the document]
Comment (sman-81) [reply maven-jxr/src/main/java/org/apache/maven/jxr/pacman/BaseType.java:58]: It's good practice to provide `toString` implementations.
The toString() method is used to represent objects as human-readable text.
Great help in debugging!

Square brackets are widely accepted (e.g. the toString template of Eclipse uses them).

> format [...] can be misunderstood as array

and curly braces can be misunderstood as Map ;)

File: maven-jxr-plugin/src/it/JXR-143_nofork/pom.xml (hunk 2 above)
[the remark this answers is not in the document]
Comment (slawekjaranowski) [reply maven-jxr-plugin/src/it/JXR-143_nofork/pom.xml:47]: Current build on CI on master pass as I see ...

rule `Always Pass` probably was added in order to override rules from parent ... I will check it.

**Review by slawekjaranowski — APPROVED** [review]

**Comment by slawekjaranowski** [comment]
@sman-81 thanks

**Comment by sman-81** [comment]
> @sman-81 thanks

Thank you, too @slawekjaranowski

**Pull request merged**